src/functions.jl

```diff
@@ -71,3 +67,10 @@ function invfldmod((q, r), divisor)
         throw(DomainError((q, r), "inverse for fldmod(x) is not defined at this point"))
     end
 end
+
+
+# check if x is of a real-Number type
+# this is not the same as `x isa Real` which immediately excludes custom Number subtypes such as unitful numbers
+# this is also not the same as isreal(x) which is true for complex numbers with zero imaginary part
+isreal_type(x::T) where {T<:Number} = real(T) == T
```

Comment:
I think it might be better if `isreal_type` returns something like `Val(true)` and `Val(false)`, since the result doesn't depend on the value of x. We should probably also have a specialized `isreal_type(@nospecialize x::Real) = Val(true)` so the compiler doesn't have to generate so many specializations if `x` is a `Real`. And we should make these `@inline`.

Reply:
Usage would just be

```julia
if isreal_type isa Val{true}
    ....
else
    ....
end
```

That's not so bad. And returning different types instead of different values should help the compiler eliminate the branch in most cases.

Reply:
Maybe we should name this `has_real_type`, to make the semantics clearer (since `x` itself will not be a real type).